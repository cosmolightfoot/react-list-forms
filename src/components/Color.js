import React from 'react';
import PropTypes from 'prop-types';
import styles from './Color.css';


export default function Color({ name, hex, rgb }) {
  return (
    <dl>
      <dt>Name:</dt>
      <dd>{name || hex}<div className={styles.display} style={{ backgroundColor: hex }}></div></dd>
        
      <dt>Hex:</dt>
      <dd>{hex}</dd>
        
      <dt>Red:</dt>
      <dd>{rgb.red}</dd>

      <dt>Green:</dt>
      <dd>{rgb.green}</dd>

      <dt>Blue:</dt>
      <dd>{rgb.blue}</dd>
    </dl>
  );
}

Color.propTypes = {
  name: PropTypes.string,
  hex: PropTypes.string.isRequired,
  rgb: PropTypes.shape({
    red: PropTypes.number.isRequired,
    green: PropTypes.number.isRequired,
    blue: PropTypes.number.isRequired
  })
};

