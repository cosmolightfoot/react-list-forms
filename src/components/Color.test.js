import React from 'react';
import { shallow } from 'enzyme';
import Color from './Color.js';

describe('Color test', () => {
  it('renders Color component', () => {
    const wrapper = shallow(<Color name="Red" hex="#FF0000" />);
    expect(wrapper).toMatchSnapshot();
  });
});
