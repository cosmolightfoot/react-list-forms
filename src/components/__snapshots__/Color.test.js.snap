// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Color test renders Color component 1`] = `
<dl>
  <dt>
    Color:
  </dt>
  <dd>
    Red
    <div
      style={
        Object {
          "backgroundColor": "#FF0000",
          "display": "inline-block",
          "height": "0.8em",
          "width": "0.8em",
        }
      }
    />
  </dd>
  <dt>
    Hex:
  </dt>
  <dd>
    #FF0000
  </dd>
  <dt>
    Red:
  </dt>
  <dd>
    255
  </dd>
  <dt>
    Green:
  </dt>
  <dd>
    0
  </dd>
  <dt>
    Blue:
  </dt>
  <dd>
    0
  </dd>
</dl>
`;
